block/validate.go

```diff
@@ -0,0 +1,89 @@
+package block
+
+import (
+	"context"
+	"errors"
+
+	"github.com/dymensionxyz/dymint/settlement"
+	"github.com/dymensionxyz/gerr-cosmos/gerrc"
+	"github.com/tendermint/tendermint/libs/pubsub"
+)
+
+// onNewStateUpdateFinalized will update the last validated height with the last finalized height
+func (m *Manager) onNewStateUpdateFinalized(event pubsub.Message) {
```

Comment:
```suggestion
// onNewStateUpdateFinalized will update the last validated height with the last finalized height
// (because there is no point validating it)
func (m *Manager) onNewStateUpdateFinalized(event pubsub.Message) {
```

Reply:
clarification added